session/macaroon.go

```diff
@@ -0,0 +1,76 @@
+package session
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"strconv"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"google.golang.org/protobuf/proto"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+	"gopkg.in/macaroon.v2"
+)
+
+const (
+	// SuperMacaroonRootKeyBitmask is the prefix bitmask we set on a super
+	// macaroon's root key to clearly mark it as such.
+	SuperMacaroonRootKeyBitmask uint64 = 0xFFEEDDCC_00000000
```

Comment:
Where's this come from? 

Couldn't we alternatively go with a prefix? Or maybe I don't understand the implications of the ID: I thought it was just a top-level key we used to read the macaroon secret to validate incoming macaroons?

Reply:
We need a way to identify a super macaroon by just reading the macaroon itself. Since a super macaroon is always baked by lnd, we can specify the _ID_ of the root key to use. And that ID (not the root key itself, obviously) is encoded within the macaroon (in a field called `storage_id`). Since the ID is `uint64`, we have 8 bytes to work with and this mask is effectively the prefix we use for any super macaroon.
I guess we could've simplified the code somewhat by generating 4 random bytes, prepending the prefix and only afterwards convert it to `uint64` instead of doing bit mask magic...